### PR TITLE
sys/fs/constfs: drop dummy implementations

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

VFS will handle it if those functions are not implemented.

### CODE REVIEW

File: sys/fs/constfs/constfs.c [hunk 1]
```diff
@@ -244,18 +203,6 @@ static ssize_t constfs_read(vfs_file_t *filp, void *dest, size_t nbytes)
...
-static ssize_t constfs_write(vfs_file_t *filp, const void *src, size_t nbytes)
-{
-    DEBUG("constfs_write: %p, %p, %lu\n", (void *)filp, src, (unsigned long)nbytes);
-    /* Read only file system */
-    DEBUG("constfs_write: read only FS\n");
-    /* prevent warning: unused parameter */
-    (void) filp;
-    (void) src;
-    (void) nbytes;
-    return -EBADF;
```
Comment (chrysn) [inline sys/fs/constfs/constfs.c:256]: This is technically a change (vfs_write without a write returns -EINVAL), but I'd say it's for the better.

**Review by chrysn — COMMENTED** [review]
Generally a good change, I went through vfs.c to verify that it all still does the same (but did not do any testing yet).

File: sys/fs/constfs/constfs.c [hunk 2]
```diff
@@ -91,28 +79,6 @@ const vfs_file_system_t constfs_file_system = {
...
-    (void) mountp; /* prevent warning: unused parameter */
-    return 0;
-}
-
-static int constfs_unlink(vfs_mount_t *mountp, const char *name)
-{
-    /* Removing files is prohibited */
-    (void) mountp; /* prevent warning: unused parameter */
-    (void) name; /* prevent warning: unused parameter */
-    return -EROFS;
```
[the remark this answers is not in the document]
Comment (benpicco) [reply sys/fs/constfs/constfs.c:113]: Shouldn't the default then rather be changed to `-EROFS`? The function not being implemented is not a permission issue.

[the remark this answers is not in the document]
Comment (chrysn) [reply sys/fs/constfs/constfs.c:113]: I have no strong opinions on the concrete standard error values we return due to POSIXish conventions; EROFS sounds just as good as EPERM to me.

[the remark this answers is not in the document]
Comment (benpicco) [reply sys/fs/constfs/constfs.c:113]: I decided to change it to `-EROFS` in VFS.

**Review by chrysn — APPROVED** [review]
Looks good, did some light testing (on rm, not full coverage, but I think it's enough).

Thanks, and ACK.

**Pull request merged**